src/Bicep.LangServer/Refactor/ExpressionAndTypeExtractor.cs

```diff
@@ -345,45 +368,131 @@ private string GetQuotedText(string text)
 
     // Finds a suitable location to create a new declaration, putting it near existing declarations of that type
     //   *above the extraction point*, if there are any.
-    private int FindOffsetToInsertNewDeclaration(CompilationContext compilationContext, int extractionOffset, Type declarationSyntaxType)
+    private (int offset, bool insertNewlineBefore, bool insertNewlineAfter) FindOffsetToInsertNewDeclaration(CompilationContext compilationContext, StatementSyntax extractionStatement, int extractionOffset, Type declarationSyntaxType)
     {
         ImmutableArray<int> lineStarts = compilationContext.LineStarts;
 
-        var extractionLine = GetPosition(extractionOffset).line;
-        var startSearchingAtLine = extractionLine - 1;
+        var extractionLine = TextCoordinateConverter.GetPosition(lineStarts, extractionOffset).line;
 
-        for (int line = startSearchingAtLine; line >= 0; --line)
```

Comment:
The switch from for to a linq statement is a perf optimization, since the old way was theoretically x^2 on program length.